### DISCUSSION TITLE
Dblink overwhelms database

### DISCUSSION THREAD

**Discussion by inamirandaz** [supabase/supabase#10680] [Questions]
Hi,

We are using the dblink extension to copy a large table (33M records, ~10GB) from one Supabase project to another. We had to chunk it in batches of 1M per query because anything larger than that fails.  However, after 10 dblink queries over a span of an hour (i.e. 10M rows inserted), the database got ovewhelmed and we got this error
`"FATAL: the database system is in recovery mode\\nFATAL: the database system is in recovery mode"`

We were able to recovery the target db by restarting it. We plan to continue with smaller chunks and the queries more spread out over time. 

Appreciate if anyone can give advice on how we can optimize the above. Or if there are better alternatives to do copy large tables/db from one Supabase project to another.

**Comment by sveetya** [comment]
Did you solve it somehow?
